### ISSUE TITLE
Enable search for not-logged users in shared folders

### ISSUE THREAD

**Issue by SlavikCA** [nextcloud/server#1794]
### Steps to reproduce
1. Share folder with files via the link
2. While not logged, as a user - open the link (or in Incognito mode)

### Expected behaviour
I expect, that search bar will be available, so I can search for files

### Actual behaviour
There is no search bar. It only shown for authenticated users.

### Server configuration
**Operating system**:
Ubuntu 16.04

**Web server:**
Apache 2.4.7

**Database:**
MySQL

**PHP version:**
7.0

**Nextcloud version:** (see Nextcloud admin page)
10.0.1

**Updated from an older Nextcloud/ownCloud or fresh install:**
Fresh install

**Where did you install Nextcloud from:**
From the https://download.nextcloud.com/server/releases/nextcloud-10.0.1.zip

**Signing status:**
<details>
<summary>Signing status</summary>

```
http://backup.slavikf.com/index.php/settings/integrity/failed
No errors have been found.
```
</details>

**List of activated apps:**
Default set of apps for clean install

**Are you using external storage, if yes which one:** local/smb/sftp/...
Only local

**Are you using encryption:** yes/no
No

**Are you using an external user-backend, if yes which one:** LDAP/ActiveDirectory/Webdav/...
No

### Client configuration
**Browser:**
Chrome 53

**Operating system:**
Windows 10

![image](https://cloud.githubusercontent.com/assets/6293679/19506235/a54e9e12-957e-11e6-90fd-3b6eedc1dfbd.png)

**Comment by fakemoth** [comment]
This is a must, it would bring up the potential of NC: if public visitors via a shared folder would be able to search we would have an awesome download manager. Just think about all the paperwork in an institution for example - how would one find something?
Also it would help if we could hide/disable the "Download" button in order for the visitors to download only specific files and not the whole directory. Or did I miss such a setting?

**Comment by leonidasv** [comment]
There's a pull request merged on Owncloud 7 for file search in shared folders. Maybe this could help to implement this function.

https://github.com/owncloud/core/pull/10287/files

**Comment by jospoortvliet** [comment]
The issue is that people then would be able to see information that they should not: the search shows all files a user has access to.

First, they are anonymous users so the don't have access to anything. You could give them access to the files of the user who shared those files but then they can see everything - not just the shared files. That is quite a security and privacy problem. You could limit it to only what is publicly shared, that would fix this, but even that is usually not what you want: I share a public link with my parents, and another with a colleague. I don't want that colleague to find the public sharing link I shared with my parents, that is none of their business!

So what exactly is this search supposed to find? Just the files in that single shared folder? That might work... But do you need search for a single share? Maybe for big ones, yes, that might be interesting, certainly a feature that could be developed if somebody is interested (or has the money to pay for it, as usual).

**Comment by SlavikCA** [comment]
Yes, your last paragraph is correct: search files just in single share folder.
Do I need it? Of course, that's what the issue is about.

**My use case:**
I'm hosting music sheets (notes) for our churches. That's thousands of files, so I need search functionality for web site visitors. And that's public share:
https://notes.propovednik.com/

Since there is no built-in NC search,  I installed Nextant. It works, but takes too much resources and seems to be overkill for the job.

**Comment by jospoortvliet** [comment]
Ok, makes sense! Then the question is who wants to work on it.

btw @leonidasv as that code went in years ago, that is certainly already part of Nextcloud ;-)

**Comment by kaysond** [comment]
Is anyone working on this? If I'm interested in contributing where would I start?

**Comment by tuxcrafter** [comment]
I would also like to see a search in shared folders, but only reaching the files within the scope of the share. Maybe an implementation with a quick index search (find output?) when first using the search and then parsing the index quickly on subsequent searches?

**Comment by fakemoth** [comment]
I see people are flocking around this :) Just wanted to say that I still think about and miss this feature, a longer explanation here on the forum https://help.nextcloud.com/t/search-in-public-shared-folders-and-option-for-disabling-the-download-button/20624

**Comment by jospoortvliet** [comment]
@kaysond you can perhaps ask @daita - he's done some search work and might be able to give some pointers on where to start?!?!

**Comment by Tanarri** [comment]
Any news here? The search at Shared-Links is a feature, which many people will miss

**Comment by skjnldsv** [comment]
We could maybe make unified search available on public pages too and only allows some search providers?
@jancborchardt ?

**Comment by agustinjm** [comment]
i need this feature -->@Tanarri, any news here? The search at Shared-Links is a feature, which many people will miss ,  best wishes

**Comment by Tanarri** [comment]
@agustinjm I haven't heard anything
btw: I am only a user of nextcloud, not in the programmer team

**Comment by Practicalbutterfly5** [comment]
Any updates on this? Recently I had shared a large folder with friends, and they had difficulty finding the file they were looking for. This would be the most important feature for my use case of Nextcloud as file server.

**Comment by JuliaGilmore** [comment]
Echoing @Practicalbutterfly5 ... are there any plans to add this functionality? We're encouraging more use of Nextcloud for file management and this has come up several times as a request from our teams.

**Comment by jospoortvliet** [comment]
@JuliaGilmore if you have a Nextcloud account manager, best ask them - that might put it on the agenda of engineering. Otherwise, it simply depends on a volunteer, as always.

**Comment by SecJoe** [comment]
Any update here? Function would be helpful for every shared folder from a certain size...
